### DISCUSSION TITLE
Limit runtime?

### DISCUSSION THREAD

**Discussion by wmertens** [community/community#25631] [Actions]
<p>I had a test that hung and I got a notification that it was killed after 6 hours.</p>
<p>This seems wasteful. Is there a way to configure a time limit for the workflows? For me, after 20 minutes I’m really not expecting anything to still be active.</p>

**Comment by wipe2238** [comment]
<p>Here you go <a href="https://help.github.com/en/articles/workflow-syntax-for-github-actions#jobsjob_idstepstimeout-minutes" rel="nofollow noopener">https://help.github.com/en/articles/workflow-syntax-for-github-actions#jobsjob_idstepstimeout-minutes</a></p>

**Comment by hugovk** [comment, marked as answer]
<p>You can add <span>timeout-minutes</span><span>:</span> 20 to your job:</p>
<ul>
<li><a href="https://help.github.com/en/articles/workflow-syntax-for-github-actions#jobsjob_idtimeout-minutes" rel="nofollow noopener">https://help.github.com/en/articles/workflow-syntax-for-github-actions#jobsjob_idtimeout-minutes</a></li>
</ul>
<p>For example:</p>
<ul>
<li><a href="https://github.com/python-pillow/Pillow/pull/4084/files" rel="nofollow noopener">https://github.com/python-pillow/Pillow/pull/4084/files</a></li>
</ul>
<p>See also:</p>
<ul>
<li><a href="https://github.community/t/cant-cancel-hung-job-failed-to-cancel-check-suite/16346">https://github.community/t5/GitHub-Actions/Can-t-cancel-hung-job-quot-Failed-to-cancel-check-suite-quot/m-p/32673</a></li>
</ul>

**Comment by venh** [comment]
<p>If I have many repos and each repo has many actions, it will be very difficult to apply this setting manually on every action. Is there an org level setting, so that the Org Admin can set it and it will get applied for each action in each repo belonging to that org? That way, the Org Admin can control it for all the repos from a central location.</p>

**Comment by schrodervictor** [comment]
<p>Indeed, it would be really great to have control about the Github Actions timeouts at the organization level, while still being able to override them on per job basis. Or maybe a general hard limit, for the whole organization and/or repository. The number of minutes consumed by a job affects billing directly, so the organization has to be able to enforce some limits there. Currently, a single mistake in a job, causing it to hang for example, can cause the exhaustion of all the available minutes.</p>
